### PR TITLE
5 thumbnail filter

### LINKED ISSUE #5: filter search by presence of certain image files
it would be nice to be able to filter blockbuilder search results to show only blocks that have (or do not have) some combination of these conventionally named image files:

`thumbnail.png`
`preview.png`

### PR DESCRIPTION
**Pull request by micahstubbs**
This PR adds client-side thumbnail filtering.  click the checkbox to show only blocks with a thumbnail image.

5-client-side-thumbnail-filter

fix #5 
see also #29 search elasticsearch for only blocks with thumbnails (the server-side solution)

### CODE REVIEW

**Comment by micahstubbs** [comment]
ok, I increased the default search results count to `512` and now tell **Elasticsearch** that we don't want the (potentially rather big) `_source` fields `code` and `readme`.  this gives us both lots of results and as fast or faster searches that before 🎉 

@enjalot will ask forgiveness for doing three things in one PR 😅   

feel like these two new things really help make the client-side image filtering a nice experience, with plenty of results to peruse even when the image filter is applied.

File: index.js [hunk 1]
```diff
@@ -174,57 +173,70 @@ function searchES(es, submittedQuery, callback) {
   }
   query.size = submittedQuery.size;
 
-  console.log(new Date(), "submitted", JSON.stringify(submittedQuery))
-  console.log(new Date(), "QUERY", JSON.stringify(query))
+  console.log(new Date(), 'submitted', JSON.stringify(submittedQuery));
```
Comment (Fasani) [inline index.js]: Is this intentional?

File: react/components/results.js [hunk 2]
```diff
@@ -21,15 +21,29 @@ const ResultsComponent = React.createClass({
...
+      );
     });
   },
   onMouseOut() {
     Mousetrap.unbind('shift');
   },
   render() {
     var query = this.props.query;
-    var results = this.props.results;
+    console.log('query', query);
```
Comment (Fasani) [inline react/components/results.js]: And Again?

Comment (micahstubbs) [reply react/components/results.js]: oh, this isn’t, intentional.  good find, will fix!

File: index.js (hunk 1 above)
Comment (micahstubbs) [reply index.js]: good question, probably not

Comment (micahstubbs) [reply index.js]: ok, looks like it's just a style change

File: react/components/results.js (hunk 2 above)
Comment (micahstubbs) [reply react/components/results.js]: ok, it looks like this is also a prettier formatting change

**Pull request merged**